packages/nextjs/src/config/loaders/dataFetchersLoader.ts

```diff
@@ -0,0 +1,107 @@
+
+  try {
+    userAST = makeAST(userCode, isTS);
+    templateAST = makeAST(templateCode, false);
+  } catch (err) {
+    logger.warn(`Couldn't add Sentry to ${filepath} because there was a parsing error: ${err}`);
+  }
+
+  // This will be true iff `makeAST` errored, but TS can't quite make the logical leap, so we check it directly
+  if (!userAST || !templateAST) {
+    // Replace the template code with an empty string, so in the end the user code is untouched
+    return [userCode, ''];
+  }
+
+  // Comments are useful to have in the template for anyone reading it, but don't make sense to be injected into user
+  // code, because they're about the template-i-ness of the template, not the code itself
+  removeComments(templateAST);
+
+  for (const fnName of DATA_FETCHING_FUNCTIONS) {
+    const matchingNodes = findIdentifiers(userAST, fnName);
+
+    // If the current function exists in a user's code, prefix all references to it with an underscore, so as not to
+    // conflict with the wrapped version we're going to create
+    if (matchingNodes.length > 0) {
+      matchingNodes.forEach(nodePath => (nodePath.node.name = `_${fnName}`));
+    }
+
+    // Otherwise, if the current function doesn't exist anywhere in the user's code, delete the code in the template
+    // wrapping that function
+    //
+    // Note: We start with all of the possible wrapper lines in the template and delete the ones we don't need (rather
+    // than starting with none and adding in the ones we do need) because it allows them to live in our souce code as
+    // *code*. If we added them in, they'd have to be strings containing code, and we'd lose all of the benefits of
+    // syntax highlighting, linting, etc.
+    else {
+      // We have to look for declarations and exports separately because when we build the SDK, Rollup turns
+      //     export const XXX = ...
+      // into
+      //     const XXX = ...
+      //     export { XXX }
+      findExports(templateAST, fnName).remove();
+      findDeclarations(templateAST, fnName).remove();
+    }
+  }
+
+  return [userAST.toSource(), templateAST.toSource()];
+}
+
+/**
+ * Wrap `getStaticPaths`, `getStaticProps`, and `getServerSideProps` (if they exist) in the given page code
+ */
+function wrapDataFetchersLoader(this: LoaderThis<LoaderOptions>, userCode: string): string {
+  // We know one or the other will be defined, depending on the version of webpack being used
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const { projectDir } = this.getOptions ? this.getOptions() : this.query!;
+
+  // For now this loader only works for ESM code
+  // TODO: Can you even write nextjs pages in CJS?
+  if (!isESM(userCode)) {
+    return userCode;
+  }
+
+  // If none of the functions we want to wrap appears in the page's code, there's nothing to do
+  if (DATA_FETCHING_FUNCTIONS.every(functionName => !userCode.includes(functionName))) {
+    return userCode;
+  }
```

Comment:
Wondering if it is a problem that we get false positives here: If `userCode` includes a function name which e.g. is commented out, we wouldn't return here. Doesn't even have to be comments. Could also be a variable identifier or whatever (e.g. `const propFromGetServerSideProps = ...`).

Reply:
You're right that we'd miss the fail-fast check, but in `wrapFunctions` we wouldn't find any identifiers with that name, and that'd cause us to pull the relevant line from the template. But it's not an unreasonable question (and one others might have), so I'll add a comment.